### ISSUE TITLE
improved docs for getting basic setup running

### ISSUE THREAD

**Issue by jjb** [lsegal/yard#1507]
thanks for a great project!

took me some trial and error to figure out dependencies

settled on this invocation

```console
apt-get update
apt-get install -y rdoc
gem install yard webrick 
yard server
```

maybe some of this could be put into the readme. i'm happy to make a PR but a maintainer might be better at choosing where to put it.

**Comment by lsegal** [comment]
This seems like a platform (and distro) specific setup, typical to Debian, that doesn't release rdoc (or webrick) alongside Ruby by default. Technically speaking, neither of those libraries are recommended unless you plan on using RDoc style formatting over Markdown, so I'm not sure this is the right approach. If we were making a recommendation, it would probably be redcarpet + puma or thin, but it very much depends on (a) whether you need RDoc formatting and (b) if you're already using some other webserver (let's face it, the vast majority of Ruby usage is via Rails and already has a server in the Gemfile).

Also worth noting that Ruby's own docs cover this somewhat, at the very least on Debian they recommend using the [ruby-full](https://www.ruby-lang.org/en/documentation/installation/#apt) package which contains rdoc (though probably not webrick). Most users are not running `yard server`, thus webrick is likely not in the standard install path and using the recommended install settings on Debian should work for most users.

YARD opts for optional dependencies in order to allow for a wide range of setups without breaking downstream usage (i.e. forcing users to pin to specific versions of webrick, rdoc, etc). The alternative would be to vend YARD as a set of packages that are command-specific, like a `gem install yard-server`, but users would need to know to install this, and separating this way would not be backward compatible. 

tl;dr this is very specific to Debian-like package ecosystems that like to break Ruby's standard install up into specific prefixed packages. It might be worth making a note that YARD expects a "standard Ruby installation", but it would be difficult to be too prescriptive about how this works given the many package managers that exist.

**Comment by jjb** [comment]
gotcha, okay!

fyi i'm on the official ruby docker image, which is based on debian, but builds its own ruby
